test/tests_for_utils/catalog_cache_test.py

```diff
@@ -0,0 +1,175 @@
+# This test only tests code that can be exercised without a network connection to services.
+# That code is tested in integration tests.
+from unittest.mock import create_autospec
+
+import pytest
+
+from execution_engine2.utils.catalog_cache import CatalogCache
+from installed_clients.CatalogClient import Catalog
+from utils_shared.test_utils import (
+    assert_exception_correct,
+    CLIENT_GROUP_CONFIG,
+)
+
+
+@pytest.fixture
+def catalog():
+    return create_autospec(Catalog, spec_set=True, instance=True)
+
+
+@pytest.fixture
+def catalog_cache():
+    return create_autospec(CatalogCache, spec_set=True, instance=True)
+
+
+def test_fail_cc():
+    with pytest.raises(ValueError) as e:
+        CatalogCache(None)
+    assert_exception_correct(
+        e.value, ValueError("Please provide instance of catalog client")
+    )
+
+    # Test that a new catalog call is made once more
+    with pytest.raises(ValueError) as e:
+        catalog_cache = CatalogCache(catalog=catalog)
+        catalog_cache.lookup_git_commit_version(method=None, service_ver="dev")
+    assert_exception_correct(e.value, ValueError("Must provide a method to lookup"))
+
+
+def assert_call_count_and_return_val(
+    mock, call_count, return_value, expected_return_value
+):
+    assert mock.call_count == call_count
+    assert return_value == expected_return_value
+
+
+def test_get_catalog(catalog):
+    assert catalog == CatalogCache(catalog).get_catalog()
+
+
+def test_cc_job_reqs(catalog):
+    """Test to see the job requirements cache is being used."""
+    test_return = {"Test1"}
+    catalog.list_client_group_configs.return_value = test_return
+    catalog_cache = CatalogCache(catalog=catalog)
+    job_reqs_cache = catalog_cache.get_job_resources_cache()
+
+    # Test Cache is called on second call
+    rv1 = catalog_cache.lookup_job_resource_requirements(
+        module_name="test1", function_name="test1"
+    )
+
+    assert catalog.list_client_group_configs.call_count == 1
+    # Test to make sure it still returns values based on the catalog
+    assert rv1 == test_return
+    assert "test1" in job_reqs_cache and "test1" in job_reqs_cache["test1"]
+    catalog.list_client_group_configs.assert_called_with(
+        {"module_name": "test1", "function_name": "test1"}
+    )
+
+    catalog.list_client_group_configs.return_value = CLIENT_GROUP_CONFIG
+    catalog_cache._method_version_cache["test1"]["test1"] = "Something else"
```

Comment:
I don't understand what this line is for

Reply:
I meant to use the job requirements cache and check for "Something else".
FIxed the test